django_ipam/base/models.py

```diff
@@ -55,6 +59,70 @@ def request_ip(self, options=None):
             return ip_address
         return None
 
+    def import_csv(self, file):
+        ipaddress_model = swapper.load_model("django_ipam", "IpAddress")
+        subnet_model = swapper.load_model("django_ipam", "Subnet")
+        if file.name.endswith(('.xls', '.xlsx')):
+            book = xlrd.open_workbook(file_contents=file.read())
+            sheet = book.sheet_by_index(0)
+            row = []
+            for row_num in range(sheet.nrows):
+                row.append(sheet.row_values(row_num))
+            reader = iter(row)
+        else:
+            reader = csv.reader(StringIO(file.read().decode('utf-8')),
+                                delimiter=',')
+        subnet_name = next(reader)[0].strip()
+        subnet_value = next(reader)[0].strip()
+        try:
+            subnet_exists = subnet_model.objects.filter(name=subnet_name, subnet=subnet_value).exists()
+        except (ValidationError, ValueError):
+            return False
+        if subnet_exists:
+            subnet = subnet_model.objects.get(name=subnet_name,
+                                              subnet=subnet_value)
+        else:
+            try:
+                subnet = subnet_model(name=subnet_name,
+                                      subnet=subnet_value)
+                subnet.full_clean()
+                subnet.save()
+            except (ValidationError, ValueError):
+                return False
+        next(reader)
+        next(reader)
+        ipaddress_list = []
+        for row in reader:
+            if not ipaddress_model.objects.filter(subnet=subnet,
+                                                  ip_address=row[0].strip(),
+                                                  description=row[1].strip()).exists():
+                instance = ipaddress_model(subnet=subnet,
+                                           ip_address=row[0].strip(),
+                                           description=row[1].strip())
+                try:
+                    instance.full_clean()
+                except (ValidationError, ValueError):
+                    return False
```

Comment:
same as before, raise the exception which is caught elsewhere to display the information about what went wrong